test/react/functional-components/safe4.js

```diff
@@ -0,0 +1,36 @@
+if (!window.__evaluatePureFunction) {
+  window.__evaluatePureFunction = function(f) {
+    return f();
+  };
+}
+
+function foo(props) {
+  var abstract = props.props;
+  var outer = {};
+  var res =__evaluatePureFunction(() => {
+    var x = {a: 1};
+    __evaluatePureFunction(() => {
+      var y = {};
+  
+      // we know `outer` won't change so we don't havoc it:
+      abstract(outer);
+  
+      // should this havoc `x` or not?
+      // my intuition is that it *shouldn't* because the inner scope is "pure".
+      // but I think this PR would change that.
```

Comment:
This comment doesn't make sense in a test.

Reply:
Lol, yeah, I should have removed that